sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -12,18 +16,52 @@ namespace core
 namespace http
 {
 
-class http_request
+// BodyStream is used to read data to/from a service
+class BodyStream
 {
+public:
+  static BodyStream& null;
```

Comment:
Reference on null is something very uncommon in C++. If it can be null, why not use pointer?

Reply:
changed to prt for now